spel/scripts/amigen8-build.sh

```diff
@@ -0,0 +1,412 @@
+#!/bin/bash
+# shellcheck disable=SC2034,SC2046
+#
+# Execute AMIGen8 scripts to prepare an EC2 instance for the AMI Create Image
+# task.
+#
+##############################################################################
+PROGNAME="$(basename "$0")"
+AMIGENBOOTSIZE="${SPEL_AMIGENBOOTSIZE:-UNDEF}"
+AMIGENBRANCH="${SPEL_AMIGENBRANCH:-master}"
+AMIGENBUILDDEV="${SPEL_AMIGENBUILDDEV:-/dev/xvda}"
+AMIGENCHROOT="${SPEL_AMIGENCHROOT:-/mnt/ec2-root}"
+AMIGENCLIV1SRC="${SPEL_AMIGENCLIV1SRC:-UNDEF}"
+AMIGENCLIV2SRC="${SPEL_AMIGENCLIV2SRC:-UNDEF}"
+AMIGENFSTYPE="${SPEL_AMIGENFSTYPE:-xfs}"
+AMIGENREPOS="${SPEL_AMIGENREPOS:-UNDEF}"
+AMIGENREPOSRC="${SPEL_AMIGENREPOSRC:-UNDEF}"
+AMIGENROOTNM="${SPEL_AMIGEN8_ROOTNM:-UNDEF}"
+AMIGENSOURCE="${SPEL_AMIGEN8SOURCE:-https://github.com/plus3it/AMIgen8.git}"
+AMIGENSSMAGENT="${SPEL_AMIGENSSMAGENT:-UNDEF}"
+AMIGENSTORLAY="${SPEL_AMIGENSTORLAY:-UNDEF}"
+AMIGENTIMEZONE="${SPEL_TIMEZONE:-UTC}"
+AMIGENVGNAME="${SPEL_AMIGENVGNAME:-UNDEF}"
+CLOUDPROVIDER="${SPEL_CLOUDPROVIDER:-aws}"
+DEBUG="${DEBUG:-UNDEF}"
+ELBUILD="/tmp/el-build"
+
+# Make interactive-execution more-verbose unless explicitly told not to
+if [[ $( tty -s ) -eq 0 ]] && [[ ${DEBUG} == "UNDEF" ]]
+then
+   DEBUG="true"
+fi
+
+
+# Error handler function
+function err_exit {
+   local ERRSTR
+   local ISNUM
+   local SCRIPTEXIT
+
+   ERRSTR="${1}"
+   ISNUM='^[0-9]+$'
+   SCRIPTEXIT="${2:-1}"
+
+   if [[ ${DEBUG} == true ]]
+   then
+      # Our output channels
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+# Run the builder-scripts
+function BuildChroot {
+   local PATHY
+
+   PATHY="${ELBUILD}/$( sed -e 's#^.*/##' -e 's/\.git.*$//' <<< "${AMIGENSOURCE}" )"
+
+   # Invoke disk-partitioner
+   bash -x "${PATHY}"/$( ComposeDiskSetupString ) || \
```

Comment:
I have a strong preference for keeping the strict-mode invocation here on each call, even though it is set in amigen8...
```suggestion
   bash -eux -o pipefail "${PATHY}"/$( ComposeDiskSetupString ) || \
```